src/real_ip.rs

```diff
@@ -0,0 +1,299 @@
+    "13.32.0.0/15",
+    "13.35.0.0/16",
+    "13.48.32.0/24",
+    "13.54.63.128/26",
+    "13.59.250.0/26",
+    "13.113.196.64/26",
+    "13.113.203.0/24",
+    "13.124.199.0/24",
+    "13.210.67.128/26",
+    "13.224.0.0/14",
+    "13.228.69.0/24",
+    "13.233.177.192/26",
+    "13.249.0.0/16",
+    "15.158.0.0/16",
+    "15.188.184.0/24",
+    "15.207.13.128/25",
+    "15.207.213.128/25",
+    "18.64.0.0/14",
+    "18.154.0.0/15",
+    "18.160.0.0/15",
+    "18.164.0.0/15",
+    "18.172.0.0/15",
+    "18.192.142.0/23",
+    "18.200.212.0/23",
+    "18.216.170.128/25",
+    "18.229.220.192/26",
+    "18.238.0.0/15",
+    "18.244.0.0/15",
+    "34.195.252.0/24",
+    "34.216.51.0/25",
+    "34.223.12.224/27",
+    "34.223.80.192/26",
+    "34.226.14.0/24",
+    "35.158.136.0/24",
+    "35.162.63.192/26",
+    "35.167.191.128/26",
+    "36.103.232.0/25",
+    "36.103.232.128/26",
+    "44.227.178.0/24",
+    "44.234.90.252/30",
+    "44.234.108.128/25",
+    "52.15.127.128/26",
+    "52.46.0.0/18",
+    "52.47.139.0/24",
+    "52.52.191.128/26",
+    "52.56.127.0/25",
+    "52.57.254.0/24",
+    "52.66.194.128/26",
+    "52.78.247.128/26",
+    "52.82.128.0/19",
+    "52.84.0.0/15",
+    "52.124.128.0/17",
+    "52.199.127.192/26",
+    "52.212.248.0/26",
+    "52.220.191.0/26",
+    "52.222.128.0/17",
+    "54.182.0.0/16",
+    "54.192.0.0/16",
+    "54.230.0.0/17",
+    "54.230.128.0/18",
+    "54.230.200.0/21",
+    "54.230.208.0/20",
+    "54.230.224.0/19",
+    "54.233.255.128/26",
+    "54.239.128.0/18",
+    "54.239.192.0/19",
+    "54.240.128.0/18",
+    "58.254.138.0/25",
+    "58.254.138.128/26",
+    "64.252.64.0/18",
+    "64.252.128.0/18",
+    "65.8.0.0/16",
+    "65.9.0.0/17",
+    "65.9.128.0/18",
+    "70.132.0.0/18",
+    "71.152.0.0/17",
+    "99.79.169.0/24",
+    "99.84.0.0/16",
+    "99.86.0.0/16",
+    "108.138.0.0/15",
+    "108.156.0.0/14",
+    "116.129.226.0/25",
+    "116.129.226.128/26",
+    "118.193.97.64/26",
+    "118.193.97.128/25",
+    "119.147.182.0/25",
+    "119.147.182.128/26",
+    "120.52.12.64/26",
+    "120.52.22.96/27",
+    "120.52.39.128/27",
+    "120.52.153.192/26",
+    "120.232.236.0/25",
+    "120.232.236.128/26",
+    "120.253.240.192/26",
+    "120.253.241.160/27",
+    "120.253.245.128/26",
+    "120.253.245.192/27",
+    "130.176.0.0/17",
+    "130.176.128.0/18",
+    "130.176.192.0/19",
+    "130.176.224.0/20",
+    "143.204.0.0/16",
+    "144.220.0.0/16",
+    "180.163.57.0/25",
+    "180.163.57.128/26",
+    "204.246.164.0/22",
+    "204.246.168.0/22",
+    "204.246.172.0/24",
+    "204.246.173.0/24",
+    "204.246.174.0/23",
+    "204.246.176.0/20",
+    "205.251.200.0/21",
+    "205.251.208.0/20",
+    "205.251.249.0/24",
+    "205.251.250.0/23",
+    "205.251.252.0/23",
+    "205.251.254.0/24",
+    "216.137.32.0/19",
+    "223.71.11.0/27",
+    "223.71.71.96/27",
+    "223.71.71.128/25",
+];
+
+static CLOUD_FRONT_NETWORKS: Lazy<Vec<IpNetwork>> = Lazy::new(|| {
```

Comment:
Using `hashbrown` or `tinyset` here on CIDR integer values  would be significant faster than an iterated string match on a `Vec`. The `cidr` crate implements these as integer types.

Reply:
I've created a benchmark for this at https://github.com/Turbo87/cidr-list-benchmark. 

These are the results on my machine:

```
cidr_check              fastest       │ slowest       │ median        │ mean          │ samples │ iters
├─ cidr_bitstring_tree  165.3 ns      │ 51.87 µs      │ 749.3 ns      │ 782.9 ns      │ 1159371 │ 1159371
│                       6.048 Mitem/s │ 19.27 Kitem/s │ 1.334 Mitem/s │ 1.277 Mitem/s │         │
├─ cidr_vec_iter        181.6 ns      │ 1.255 µs      │ 197.2 ns      │ 201.8 ns      │ 133017  │ 4256544
│                       5.506 Mitem/s │ 796.2 Kitem/s │ 5.07 Mitem/s  │ 4.954 Mitem/s │         │
├─ ip_network_table     26 ns         │ 378.5 ns      │ 30.57 ns      │ 31.38 ns      │ 123100  │ 15756800
│                       38.44 Mitem/s │ 2.641 Mitem/s │ 32.71 Mitem/s │ 31.86 Mitem/s │         │
├─ ip_network_vec_iter  115.1 ns      │ 1.399 µs      │ 124.3 ns      │ 127.8 ns      │ 193559  │ 6193888
│                       8.68 Mitem/s  │ 714.7 Kitem/s │ 8.043 Mitem/s │ 7.821 Mitem/s │         │
╰─ ipnetwork_vec_iter   72.88 ns      │ 425.7 ns      │ 76.8 ns       │ 79.4 ns       │ 139257  │ 8912448
                        13.72 Mitem/s │ 2.348 Mitem/s │ 13.01 Mitem/s │ 12.59 Mitem/s │         │
```

As the results show, we could speed it up by about 1.5x via https://crates.io/crates/ip_network_table, but given the median time of 76ns for the current implementation I'm not sure it's worth the extra dependency and build time 😅